### DISCUSSION TITLE
Make Generic Version of OrderedDictionary

### DISCUSSION THREAD

**Discussion by NagayamaToshiaki** [dotnet/csharplang#5493] [Language Ideas] [answered]
I don't know why the OrderedDictionary doesn't have its generic counterpart.
Since you should avoid using non-generic collections, it makes no sense.

**Comment by iam3yal** [comment]
Have you checked out [SortedDictionary](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.sorteddictionary-2?view=net-6.0)? 

Anyhow, this is the C# language design repo for the BCL and API you should post it at https://github.com/dotnet/runtime iirc.

**Reply by svick** [reply]
@eyalalonn As [the documentation of `OrderedDictionary` explains](https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.ordereddictionary?WT.mc_id=DOP-MVP-5002735#remarks), `SortedDictionary` is not the same thing:

> The elements of an `OrderedDictionary` are not sorted by the key, unlike the elements of a `SortedDictionary<TKey,TValue>` class.

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/runtime/issues/24826.
